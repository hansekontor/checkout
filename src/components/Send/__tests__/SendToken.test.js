import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { theme } from '@assets/styles/theme';
import SendToken from '@components/Send/SendToken';
import {
    walletWithBalancesAndTokensWithCorrectState,
} from '../../Wallet/__mocks__/walletAndBalancesMock';
import { BrowserRouter as Router } from 'react-router-dom';

let realUseContext;
let useContextMock;

beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();

    // Mock method not implemented in JSDOM
    // See reference at https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
    Object.defineProperty(window, 'matchMedia', {
        writable: true,
        value: jest.fn().mockImplementation(query => ({
            matches: false,
            media: query,
            onchange: null,
            addListener: jest.fn(), // Deprecated
            removeListener: jest.fn(), // Deprecated
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
            dispatchEvent: jest.fn(),
        })),
    });
});

afterEach(() => {
    React.useContext = realUseContext;
});

test('Wallet with BCH balances and tokens and state field', () => {
    useContextMock.mockReturnValue(walletWithBalancesAndTokensWithCorrectState);
    const component = renderer.create(
        <ThemeProvider theme={theme}>
            <Router>
                <SendToken
                    tokenId={
                        'bd1acc4c986de57af8d6d2a64aecad8c30ee80f37ae9d066d758923732ddc9ba'
                    }
                />
            </Router>
        </ThemeProvider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Without wallet defined', () => {
    useContextMock.mockReturnValue({
        wallet: {},
        loading: false,
    });
    expect(() => {
        renderer.create(
            <ThemeProvider theme={theme}>
                <Router>
                    <SendToken
                        tokenId={
                            'bd1acc4c986de57af8d6d2a64aecad8c30ee80f37ae9d066d758923732ddc9ba'
                        }
                    />
                </Router>
            </ThemeProvider>,
        );
    }).toThrow('balance');
});
