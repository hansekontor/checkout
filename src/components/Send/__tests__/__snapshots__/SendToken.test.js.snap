// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet with BCH balances and tokens and state field 1`] = `
Array [
  <div
    className="sc-cSHVUG lalYdv"
  >
    6.001e-9
     
    TBS
  </div>,
  <div>
    <div
      className="ant-alert ant-alert-warning ant-alert-with-description"
      data-show={true}
      role="alert"
      style={
        Object {
          "marginBottom": "12px",
        }
      }
    >
      <span
        aria-label="exclamation-circle"
        className="anticon anticon-exclamation-circle ant-alert-icon"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="exclamation-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
          />
          <path
            d="M464 688a48 48 0 1096 0 48 48 0 10-96 0zm24-112h48c4.4 0 8-3.6 8-8V296c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8z"
          />
        </svg>
      </span>
      <div
        className="ant-alert-content"
      >
        <div
          className="ant-alert-description"
        >
          If you would like to request an icon for an eToken that has already been created, please email icons@e.cash.
        </div>
      </div>
    </div>
  </div>,
  <div
    className="ant-row"
    style={Object {}}
    type="flex"
  >
    <div
      className="ant-col ant-col-24"
      style={Object {}}
    >
      <form
        className="ant-form ant-form-horizontal"
        onReset={[Function]}
        onSubmit={[Function]}
        style={
          Object {
            "width": "auto",
          }
        }
      >
        <div
          className="sc-VigVT ktfrOp"
        >
          <div
            className="ant-row ant-form-item"
            style={Object {}}
          >
            <div
              className="ant-col ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <span
                    className="ant-input-group-wrapper"
                  >
                    <span
                      className="ant-input-wrapper ant-input-group"
                    >
                      <span
                        className="ant-input-affix-wrapper"
                        onMouseUp={[Function]}
                        style={null}
                      >
                        <span
                          className="ant-input-prefix"
                        >
                          <span
                            aria-label="wallet"
                            className="anticon anticon-wallet sc-htpNat fLtATP"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="wallet"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 464H528V448h312v128zm0 264H184V184h656v200H496c-17.7 0-32 14.3-32 32v192c0 17.7 14.3 32 32 32h344v200zM580 512a40 40 0 1080 0 40 40 0 10-80 0z"
                              />
                            </svg>
                          </span>
                        </span>
                        <input
                          autoComplete="off"
                          className="ant-input"
                          name="address"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          placeholder="eToken Address"
                          required={true}
                          style={null}
                          type="text"
                          value=""
                        />
                      </span>
                      <span
                        className="ant-input-group-addon"
                      >
                        <span
                          className="sc-iwsKbI cVxKUE"
                          onClick={[Function]}
                        >
                          <span
                            aria-label="qrcode"
                            className="anticon anticon-qrcode sc-bxivhb ctLIog"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="qrcode"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M468 128H160c-17.7 0-32 14.3-32 32v308c0 4.4 3.6 8 8 8h332c4.4 0 8-3.6 8-8V136c0-4.4-3.6-8-8-8zm-56 284H192V192h220v220zm-138-74h56c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm194 210H136c-4.4 0-8 3.6-8 8v308c0 17.7 14.3 32 32 32h308c4.4 0 8-3.6 8-8V556c0-4.4-3.6-8-8-8zm-56 284H192V612h220v220zm-138-74h56c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm590-630H556c-4.4 0-8 3.6-8 8v332c0 4.4 3.6 8 8 8h332c4.4 0 8-3.6 8-8V160c0-17.7-14.3-32-32-32zm-32 284H612V192h220v220zm-138-74h56c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm194 210h-48c-4.4 0-8 3.6-8 8v134h-78V556c0-4.4-3.6-8-8-8H556c-4.4 0-8 3.6-8 8v332c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V644h78v102c0 4.4 3.6 8 8 8h190c4.4 0 8-3.6 8-8V556c0-4.4-3.6-8-8-8zM746 832h-48c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zm142 0h-48c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
                              />
                            </svg>
                          </span>
                        </span>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
              <div
                className="ant-form-item-explain ant-form-item-explain-connected"
              >
                <div
                  className=""
                  role="alert"
                >
                  
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-VigVT ktfrOp"
        >
          <div
            className="ant-row ant-form-item"
            style={Object {}}
          >
            <div
              className="ant-col ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <span
                    className="ant-input-group-wrapper"
                  >
                    <span
                      className="ant-input-wrapper ant-input-group"
                    >
                      <span
                        className="ant-input-affix-wrapper"
                        onMouseUp={[Function]}
                        style={null}
                      >
                        <span
                          className="ant-input-prefix"
                        />
                        <input
                          className="ant-input"
                          name="value"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          placeholder="Amount"
                          required={true}
                          step={1e-9}
                          style={null}
                          type="number"
                          value=""
                        />
                        <span
                          className="ant-input-suffix"
                        >
                          TBS
                        </span>
                      </span>
                      <span
                        className="ant-input-group-addon"
                      >
                        <span
                          className="sc-jTzLTM PEluH"
                          disabled={false}
                          onClick={[Function]}
                        >
                          max
                        </span>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
              <div
                className="ant-form-item-explain ant-form-item-explain-connected"
              >
                <div
                  className=""
                  role="alert"
                >
                  
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "paddingTop": "12px",
            }
          }
        >
          <button
            className="sc-gzVnrw wXhZP"
            onClick={[Function]}
          >
            Send 
            TestBits
          </button>
        </div>
      </form>
      <div
        className="ant-descriptions ant-descriptions-bordered"
      >
        <div
          className="ant-descriptions-header"
        >
          <div
            className="ant-descriptions-title"
          >
            Token info for "TestBits"
          </div>
        </div>
        <div
          className="ant-descriptions-view"
        >
          <table>
            <tbody>
              <tr
                className="ant-descriptions-row"
              >
                <th
                  className="ant-descriptions-item-label"
                  colSpan={1}
                  style={Object {}}
                >
                  <span>
                    Token ID
                  </span>
                </th>
                <td
                  className="ant-descriptions-item-content"
                  colSpan={1}
                  style={Object {}}
                >
                  <span>
                    bd1acc4c986de57af8d6d2a64aecad8c30ee80f37ae9d066d758923732ddc9ba
                  </span>
                </td>
              </tr>
              <tr
                className="ant-descriptions-row"
              >
                <th
                  className="ant-descriptions-item-label"
                  colSpan={1}
                  style={Object {}}
                >
                  <span>
                    Decimals
                  </span>
                </th>
                <td
                  className="ant-descriptions-item-content"
                  colSpan={1}
                  style={Object {}}
                >
                  <span>
                    9
                  </span>
                </td>
              </tr>
              <tr
                className="ant-descriptions-row"
              >
                <th
                  className="ant-descriptions-item-label"
                  colSpan={1}
                  style={Object {}}
                >
                  <span>
                    Document URI
                  </span>
                </th>
                <td
                  className="ant-descriptions-item-content"
                  colSpan={1}
                  style={Object {}}
                >
                  <span>
                    https://thecryptoguy.com/
                  </span>
                </td>
              </tr>
              <tr
                className="ant-descriptions-row"
              >
                <th
                  className="ant-descriptions-item-label"
                  colSpan={1}
                  style={Object {}}
                >
                  <span>
                    Document Hash
                  </span>
                </th>
                <td
                  className="ant-descriptions-item-content"
                  colSpan={1}
                  style={Object {}}
                >
                  <span>
                    
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>,
]
`;
